Gulpfile.ts

```diff
@@ -759,23 +759,53 @@ gulp.task(nodeServerOutFile, false, [servicesFile], () => {
         .pipe(gulp.dest(path.dirname(nodeServerOutFile)));
 });
 
+import convertMap = require("convert-source-map");
+import sorcery = require("sorcery");
+declare module "convert-source-map" {
+    export function fromSource(source: string, largeSource?: boolean): SourceMapConverter;
+}
+
 gulp.task("browserify", "Runs browserify on run.js to produce a file suitable for running tests in the browser", [servicesFile], (done) => {
     const settings: tsc.Settings = getCompilerSettings({
-        outFile: "built/local/bundle.js"
+        outFile: "built/local/bundle.js",
     }, /*useBuiltCompiler*/ true);
     return gulp.src(harnessSources)
         .pipe(newer("built/local/bundle.js"))
         .pipe(sourcemaps.init())
         .pipe(tsc(settings))
         .pipe(through2.obj((file, enc, next) => {
-            browserify(intoStream(file.contents))
+            const originalMap = file.sourceMap;
+            const prebundledContent = file.contents.toString();
+            originalMap.sources = originalMap.sources.map(path.resolve); // Make paths absolute to help sorcery deal with all the terrible paths being thrown around
+            originalMap.file = "built/local/_stream_0.js"; // intoStream (below) makes browserify think the input file is named this, so this is what it puts in the sourcemap
+
+            browserify(intoStream(file.contents), { debug: true })
                 .bundle((err, res) => {
                     // assumes file.contents is a Buffer
-                    file.contents = res;
+                    const maps = JSON.parse(convertMap.fromSource(res.toString(), /*largeSource*/true).toJSON());
+                    delete maps.sourceRoot;
+                    for (let i = 0; i < maps.sources.length; i++) {
+                        if (maps.sources[i] === "_stream_0.js") maps.sources[i] = path.resolve("built/local/_stream_0.js");
+                        else maps.sources[i] = path.resolve(maps.sources[i]);

```

Comment:
another nit: these two lines would be more C-like if you'd do `maps.source[i] = maps.source[i] === '_stream_0.js' ? ...`.

Right now they look like they were written in an ML-family language.